ehrql/tables/beta/tpp.py

```diff
@@ -586,129 +653,237 @@ class occupation_on_covid_vaccine_record(EventFrame):
 
 @table
 class opa(EventFrame):
+    """
+    Outpatient appointments data is provided via the NHS Secondary Uses Service.
+
+    This table gives core details of outpatient appointments.
+    """
+
     opa_ident = Series(
         int,
         constraints=[Constraint.NotNull()],
-        description="TODO",
+        description="Unique identifier for the appointment used across the OPA tables.",
     )
     appointment_date = Series(
         datetime.date,
-        description="TODO",
+        description="The date of an appointment.",
     )
     attendance_status = Series(
         str,
-        description="TODO",
+        description=(
+            "Indicates whether or not an appointment for a care contact took place. "
+            "If the appointment did not take place it also indicates whether or not advanced warning was given. "
+            "Refer to the [NHS Data Model and Dictionary entry for attendance status]"
+            "(https://www.datadictionary.nhs.uk/data_elements/attendance_status.html).\n"
```

Comment:
question: I added a reference link here, and duplicated the permitted codes. Should we just provide the link instead?